File: 최재훈/05M05W/프로그래머스_LV1-폰켓몬/source.py
def solution(nums):
    answer = 0

    phoneMon = set()
    for num in nums:
        phoneMon.add(num)

    N = int(nums.__len__() / 2)
    phonCnt = phoneMon.__len__()

    if N == phonCnt:
        answer = phonCnt
    elif N < phonCnt:
        answer = N
    elif N > phonCnt:
        answer = phonCnt

    print(N, phoneMon.__len__(), answer)
    return answer

nums = [3, 1, 2, 3]
solution(nums)

nums = [3,3,3,2,2,4]
solution(nums)

nums = [3,3,3,2,2,2]
solution(nums)

File: 최재훈/05M05W/프로그래머스_LV1-음양 더하기/source.cpp
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// absolutes_len은 배열 absolutes의 길이입니다.
// signs_len은 배열 signs의 길이입니다.
int solution(int absolutes[], size_t absolutes_len, bool signs[], size_t signs_len) {
    int answer = 123456789;

    int sum = 0;
    for (int idx = 0; idx < absolutes_len; idx++)
    {
        int num = absolutes[idx];
        const bool bSign = signs[idx];
        if (false == bSign)
            num *= -1;

        sum += num;
    }
    answer = sum;

    return answer;
}